itis.playground/Pages/homework.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,523 @@
+        // 6 - do nothing
+        switch die {
+            case 1...3:
+                if opponent.superPower == .shield {
+                    print("\(opponent.nick) shielded super power attack from \(nick)")
+                } else {
+                    attackWithSuperpower(opponent)
+                }
+            case 4...5:
+                opponent.health -= damage
+                printAttack(nil,opponent.nick)
+                
+            default:
+                printAttack()
+        }
+    }
+    
+    
+    private func printAttack(_ action: String? = nil,
+                             _ opponent: String? = nil,
+                             with: String? = nil) {
+        var answer: String = ""
+
+        if opponent == nil {
+            answer = "\(nick) slept off and didn't attack anyone"
+        } else if action == nil {
+            answer = "\(nick) attacked \(opponent!)"
+        } else {
+            answer = "\(nick) \(action!) \(opponent!)"
+        }
+        if with != nil {
+            answer += " with \(with!)"
+        }
+        
+        print(answer)
+    }
+    
+    
+    private func attackWithSuperpower(_ opponent: AnyStudent) {
+        guard let opponents = opponent.division?.students else { return }
+        guard let teammates = division?.students else { return }
+        
+        switch superPower {
+            case .speed:
+                for player in opponents {
+                    player.health -= damage
+                }
+                printAttack("super power attacked all", "\(opponent.division!.name)", with: "⚡️")
+                
+            case .strength:
+                opponent.health -= (damage * 2)
+                printAttack("super power attacked",opponent.nick, with: "super strength")
+            case .luck:
+                health = 100
+                opponent.health -= damage
+                print("\(name) used his luck super power")
+            case .phoenix:
+                opponent.health -= damage
+                for player in teammates {
+                    player.health += 20
+                }
+                print("\(name) increased his teammates health by +20♻️")
+            case .snake:
+                opponent.health -= 20
+                for player in teammates {
+                    player.health -= 20
+                }
+                health *= 2
+                printAttack("super power attack", "his own team mates and \(opponent.nick)", with: "snake superpower")
+            case .godspeed:
+                for player in opponents {
+                    player.health -= damage*2
+                }
+                printAttack("super power attacked", "\(opponent.division!.name)", with: "GodSpeed 🔱")
+            default:
+                opponent.health -= damage
+                printAttack(nil, opponent.nick)
+                
+        }
+    }
+    
+}
+
+protocol Division {
+    
+    var name: String { get }
```

Comment:
```suggestion
    let name: String { get }
```
не меняется, сделай константой

Reply:
для протоколов "let" запрещено, линтер сказал мне использовать только { get } вместо { get, set } для переменных, которые не будут изменены